src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallLocalInstaller.cs

```diff
@@ -33,20 +36,15 @@ internal class ToolInstallLocalInstaller
             _sources = parseResult.GetValue(ToolInstallCommandParser.AddSourceOption);
             _verbosity = Enum.GetName(parseResult.GetValue(ToolInstallCommandParser.VerbosityOption));
 
-            if (toolPackageInstaller == null)
-            {
-                (IToolPackageStore,
-                    IToolPackageStoreQuery,
-                    IToolPackageInstaller installer) toolPackageStoresAndInstaller
-                        = ToolPackageFactory.CreateToolPackageStoresAndInstaller(
-                            additionalRestoreArguments: parseResult.OptionValuesToBeForwarded(ToolInstallCommandParser.GetCommand()));
-                _toolPackageInstaller = toolPackageStoresAndInstaller.installer;
-            }
-            else
-            {
-                _toolPackageInstaller = toolPackageInstaller;
-            }
-
+            (IToolPackageStore store,
+                IToolPackageStoreQuery,
+                IToolPackageDownloader installer) toolPackageStoresAndDownloader
```

Comment:
Nit: rename this to `downloader`.

```suggestion
                IToolPackageDownloader downloader) toolPackageStoresAndDownloader
```